### ISSUE TITLE
Network zones; possibly by default

### ISSUE THREAD

**Issue by jonct** [Jip-Hop/jailmaker#221]
### Proposition

Should Jailmaker default to creating new jails with `--network-zone=jlmkr` unless otherwise specified?

### Rationale

Theoretically, that would create a `vz-jlmkr` bridge on-the-fly when referenced by at least one active jail. It would then dispose of that bridge on-the-fly whenever the active-jail refcount drops back to zero. 

Out-of-the-box `/usr/lib/systemd/network/80-container-vz.network` on the host sets up DHCP and IPv4/IPv6 NAT forwarding on behalf of participating jails. Importantly: *without* any loose pre/post scripted side-effects to the host.

Meanwhile all participating jails in the same zone could reach each other, and reach the outside world. (I'll say again: theoretically.)

### Implications

I'll still need a router/dnsmasq template for my external netboot clients. But that template might simplify down into attaching a hardware interface to a zone and installing/configuring only dnsmasq.

Similarly: multiple bridges might still need scripted host side-effects. That is, unless testing shows us that nspawn respects multiple `--network-zone=` arguments producing multiple refcounted zones.

But for that new/casual user ***default***… might a standard zone be the smartest/easiest catch-all?

Kinda seems too easy… 🧐

**Comment by Jip-Hop** [comment]
I agree zones sound tempting. But I don't think it simplifies things.

> Each container may only be part of one zone

This means only one `--network-zone=` flag can be added per jail.

But more importantly none of the services running on jails in zones will be accessible from other devices on the LAN on which the SCALE NAS is located. The user would have to setup port forwarding for this:

> If private networking is enabled, maps an IP port on the host onto an IP port on the container. Takes a protocol specifier (either "tcp" or "udp"), separated by a colon from a host port number in the range 1 to 65535, separated by a colon from a container port number in the range from 1 to 65535. The protocol specifier and its separating colon may be omitted, in which case "tcp" is assumed. The container port number and its colon may be omitted, in which case the same port as the host port is implied. This option is only supported if private networking is used, such as with --network-veth, --network-zone= --network-bridge=.

But port 443 is already taken by SCALE so there's another hoop to jump through.

I think a one time setup of a bridge interface will provide the most versatile and easy to use setup.

Reference: https://manpages.debian.org/bookworm/systemd-container/systemd-nspawn.1.en.html

**Comment by Lockszmith-GH** [comment]
> But port 443 is already taken by SCALE so there's another hoop to jump through.

The way I solve this is by assigning an additional IP to the bridge interface, and having SCALE listed on that IP instead of `0.0.0.0`

**Comment by Jip-Hop** [comment]
Personally I think port forwarding in 2 places (jailmaker and also in docker-compose) would be annoying and means modifying the jail config more often (which would require a jail restart etc.). Not great if you ask me. I toyed with the idea of a default zone instead of the current default of host networking. But this is also a bit difficult to implement nicely because it contradict the systemd-nspawn default of using host networking when no networking related flags are present. So if I were to use a default zone, when no networking flags are present, then how would one specify the desire to use host networking? That would mean inventing another flag to bring back host networking...

In summary I rather stay close to nspawn default behavior when I can. If you think docs or onboarding could be improved, all help is welcome!

**Comment by Jip-Hop** [comment]
> br0: Install software and look up how to tweak that tool's configuration to avoid shifting conflicts/interactions with host services and other jails.

I don't get what you mean here. When using bridge networking there aren't any conflicts I am aware of.

**Comment by Lockszmith-GH** [comment]
Bringing my voice as just another user.

The way I see it, at this point of time - what Jailmaker introduced is an easy way to manage a pool hosted OS - including a docker host, on the immutable TrueNAS SCALE.
Yes, there are security risks with running in such a way - but that's why it should only be a layer of management of the infrastructure that will be managed, and not the infrastructure itself.

Nspawn is still installed and can be used directly if one would want to maintain it extensively.
I feel like jailmaker wouldn't benefit the community by being a complete nspawn solution, but should focus on straight forward bootstrapping as it has been doing.

Documentation is key here, as knowledge is power, and that's where the versatility comes into play.

I would venture to promote rootless-podman or NixOS based configurations inside jails as systematic solutions which can allow managing of proper security from within the hosted OS.

Is there an overhead? yes - but if we didn't want that overhead, we'd install another Linux distro, and not TrueNAS SCALE.

**Comment by Lockszmith-GH** [comment]
> I had kinda hoped that this issue would stick around longer

For that my friend - you should have opened a discussion 🧐 (See #135)

**Comment by Lockszmith-GH** [comment]
> I had kinda hoped that this issue would stick around longer

For that my friend - you should have opened a discussion 🧐 (See #135)

@Jip-Hop can you please convert this issue to a discussion?
